presto-raptorx/src/main/java/com/facebook/presto/raptorx/storage/StorageTypeConverter.java

```diff
@@ -0,0 +1,89 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.raptorx.storage;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.type.ArrayType;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.spi.type.TypeSignatureParameter;
+import com.google.common.collect.ImmutableList;
+
+import static com.facebook.presto.raptorx.util.Types.getElementType;
+import static com.facebook.presto.raptorx.util.Types.getKeyType;
+import static com.facebook.presto.raptorx.util.Types.getValueType;
+import static com.facebook.presto.raptorx.util.Types.isArrayType;
+import static com.facebook.presto.raptorx.util.Types.isMapType;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.Chars.isCharType;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.RealType.REAL;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TimeType.TIME;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.Varchars.isVarcharType;
+import static java.util.Objects.requireNonNull;
+
+public class StorageTypeConverter
+{
+    private final TypeManager typeManager;
+
+    public StorageTypeConverter(TypeManager typeManager)
+    {
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+    }
+
+    public Type toStorageType(Type type)
+    {
+        if (type.equals(BOOLEAN) ||
+                type.equals(BIGINT) ||
+                type.equals(INTEGER) ||
+                type.equals(SMALLINT) ||
+                type.equals(TINYINT) ||
+                type.equals(DOUBLE) ||
+                type.equals(REAL) ||
+                isCharType(type) ||
+                isVarcharType(type) ||
+                type.equals(VARBINARY)) {
+            return type;
+        }
+        if (type.equals(DATE) || type.equals(TIME)) {
+            return INTEGER;
+        }
+        if (type.equals(TIMESTAMP)) {
+            return BIGINT;
+        }
+        if (isArrayType(type)) {
+            return new ArrayType(toStorageType(getElementType(type)));
+        }
+        if (isMapType(type)) {
+            return mapType(toStorageType(getKeyType(type)), toStorageType(getValueType(type)));
+        }
+        throw new PrestoException(NOT_SUPPORTED, "Type not supported: " + type);
```

Comment:
"highker on Feb 4  Member
What about row types and some other types?

  @zhaotao1986
zhaotao1986 on Feb 14  Member
The big concern is, it should be backward compatible with RaptorV1. The only thing missing here is DecimalType."

This is not something we can omit, though it doesn't affect the current running. I added DecimalType, so it's backward compatible with V1